test: Use deepcopy for constant IPV6_ADDRESSES

cathay4t opened the pull request:
The unit test will fail when ran by `pytest *`. This is caused by test
case `test_state_for_verify_remove_link_local_address` changed the value
of `IPV6_ADDRESSES` which will fail `TestIfaces.test_mark_iface_as_absent.

Changed to `get_test_ipv4_addrs()` and `get_test_ipv6_addrs()` to
generate new dict at runtime.

EdDev commented on tests/lib/ifaces/base_iface_test.py:
```diff
@@ -127,7 +127,7 @@ def test_state_for_verify_remove_link_local_address(self):
         ipv6_info = {
             InterfaceIPv6.ENABLED: True,
-            InterfaceIPv6.ADDRESS: IPV6_ADDRESSES,
+            InterfaceIPv6.ADDRESS: deepcopy(IPV6_ADDRESSES),
```
It seems like the problem is with `IPV6_ADDRESSES` not being constant, or better said, not being immutable.
How about changing it to a function that returns ipv6 addresses? That will be safer for the future as well.

cathay4t commented:
New patch uploaded.

Changed to use function instead of constant.

EdDev commented on tests/lib/testlib/constants.py:
```diff
@@ -52,3 +54,11 @@
 PORT2_IFACE_NAME = "port2"
+
+
+def get_test_ipv4_addrs():
```
I am not clear why you need the private module variable at all.
IMO you can also keep the naming convention the same as the other constants here which are, i.e. `ipv4_addresses()`. 

```
def ipv4_addresses():
    return [...]
```

Just an opinionated suggestion.

cathay4t committed:
test: Use deepcopy for constant IPV6_ADDRESSES and IPV4_ADDRESSES

The unit test will fail when ran by `pytest *`. This is caused by test
case `test_state_for_verify_remove_link_local_address` changed the value
of `IPV6_ADDRESSES` which will fail `TestIfaces.test_mark_iface_as_absent.

Changed to `get_test_ipv4_addrs()` and `get_test_ipv6_addrs()` to
generate new dict at runtime.

Signed-off-by: Gris Ge <fge@redhat.com>

tyll commented on tests/lib/ifaces/base_iface_test.py:
```diff
@@ -127,7 +127,7 @@ def test_state_for_verify_remove_link_local_address(self):
         ipv6_info = {
             InterfaceIPv6.ENABLED: True,
-            InterfaceIPv6.ADDRESS: IPV6_ADDRESSES,
+            InterfaceIPv6.ADDRESS: get_test_ipv6_addrs(),
```
seems to me like ipv6_addresses should be a fixture, what do you think?

cathay4t commented:
Closing. Don't have time for this trivial issue.

cathay4t commented:
Issue is tracking at https://github.com/nmstate/nmstate/issues/1407 in case anyone has interest.
